### PR TITLE
Ss/mcubench

### PR DESCRIPTION
**Pull request by sudhakar-sah**
* Added MCUbench related files
* Updated [yolo.py](https://github.com/Deeplite/deeplite_torch_zoo/api/models/object_detection/yolo.py) to include MCUBench models and it can be accessed like other previous models 
* [mcubench](https://github.com/Deeplite/deeplite-torch-zoo/tree/ss/mcubench/results/mcubench) folder has all the files needed to recreate MCUBench. 
* Every mcubench file has instructions for usage.

### CODE REVIEW

File: deeplite_torch_zoo/api/models/object_detection/yolo.py [hunk 1]
```diff
@@ -128,7 +226,7 @@ def create_yolo_model(
 
 
 def get_model_scales(_model_key):
-    scale_dict = CUSTOM_MODEL_SCALES.get(_model_key, DEFAULT_MODEL_SCALES)
```
Comment (alexh-deeplite) [inline deeplite_torch_zoo/api/models/object_detection/yolo.py:131]: This line overwrites `CUSTOM_MODEL_SCALES` for the generation of non-mcu YOLO variants. We still want to use  `CUSTOM_MODEL_SCALES` when we call `get_model_scales` in line [239](https://github.com/Deeplite/deeplite-torch-zoo/blob/1b43d63c23d57b331f610bcd9589319a9f2f2670/deeplite_torch_zoo/api/models/object_detection/yolo.py#L239).

Simple fix is to pass the SCALES as an argument to `get_model_scales`

File: results/mcubench/benchmark_search_space_models.py [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+
+Usage : 
+    python benchmark_search_space_models.py \
+            --data_path /public-shared/mcubench/mcubench_models/ \
+            --dataset voc \
+            --hw_name nucleo-h743zi \
+            --imgsz 192 \
+            --csv_path /public-shared/mcubench/models_list/search_space_models
+            
+    python benchmark_search_space_models.py \
```
Comment (alexh-deeplite) [inline results/mcubench/benchmark_search_space_models.py:13]: remove duplicate

File: results/mcubench/benchmark_search_space_models.py [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+
+from pathlib import Path
+import pandas as pd 
+import csv
+import oapackage
+import argparse 
+import csv
+import json               
+
+with open("/public-shared/mcubench/models_list/search_space_models/search_space_models.json") as f:
```
Comment (alexh-deeplite) [inline results/mcubench/benchmark_search_space_models.py:35]: remove hard coded path

File: results/mcubench/benchmark_search_space_models.py [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+Usage : 
+    python benchmark_search_space_models.py \
+            --data_path /public-shared/mcubench/mcubench_models/ \
+            --dataset voc \
+            --hw_name nucleo-h743zi \
+            --imgsz 192 \
+            --csv_path /public-shared/mcubench/models_list/search_space_models
+            
+    python benchmark_search_space_models.py \
+        --data_path /public-shared/mcubench/mcubench_models/ \
```
Comment (alexh-deeplite) [inline results/mcubench/benchmark_search_space_models.py:14]: remove path

File: results/mcubench/benchmark_search_space_models.py [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+                # print (f'{model_name} benchmark does not exist')
+
+
+    return 1
+    
+
+
+def parse_opt(known=False):
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--data_path', type=str, default='/public-shared/mcubench/mcubench_models/', help='model and benchmark path')
```
Comment (alexh-deeplite) [inline results/mcubench/benchmark_search_space_models.py:175]: remove default path

File: results/mcubench/benchmark_search_space_models.py [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+        print (f'VERIFYING : Generated {len(data)} benchmark results for {opt.hw_name}')
+    
+    print (f'{100*"-"}')  
+    return
+     
+if __name__ == "__main__": 
+    opt = parse_opt()
+    main(opt)
+    
+    # models_path = Path("/public-shared/mcubench/mcubench_models/448/checkpoints/voc/")
```
Comment (alexh-deeplite) [inline results/mcubench/benchmark_search_space_models.py:217]: remove old code

File: results/mcubench/benchmark_first_pareto_models.py [hunk 7]
```diff
@@ -0,0 +1,240 @@
...
+                    pass
+                    # print (f'{model_name} benchmark does not exist for {imgsz}')
+
+
+    return 1
+
+# python benchmark_pareto.py --benchmarks_path /public-shared/mcubench/mcubench_models/pareto/estimated/ --models_path /public-shared/mcubench/mcubench_models/ --dataset voc --hw_name nucleo-h743zi  --csv_path ./
+
+        
+# python benchmark_first_pareto_models.py --benchmarks_path /public-shared/mcubench/mcubench_models/pareto/estimated/ --models_path /public-shared/mcubench/mcubench_models/ --dataset voc --hw_name nucleo-h743zi  --csv_path /public-shared/mcubench/models_list/first_pareto_models
```
Comment (alexh-deeplite) [inline results/mcubench/benchmark_first_pareto_models.py:179]: remove

File: results/mcubench/benchmark_search_space_models.py (hunk 3 above)
Comment (alexh-deeplite) [reply results/mcubench/benchmark_search_space_models.py:35]: can just add a local symlink to the desired directory adn put the path to the symlink here

File: results/mcubench/one_for_all_mcubench_final.sh
Comment (alexh-deeplite) [inline results/mcubench/one_for_all_mcubench_final.sh:1]: replace hard coded paths with local symlinks or cli args

**Review by alexh-deeplite — CHANGES_REQUESTED** [review]
one functional issue, the rest is all style

I will merge develop to this branch so we can see the build passing before merge

File: deeplite_torch_zoo/api/models/object_detection/yolo.py (hunk 1 above)
Comment (sudhakar-sah) [reply deeplite_torch_zoo/api/models/object_detection/yolo.py:131]: Fixed.

File: results/mcubench/benchmark_search_space_models.py (hunk 3 above)
Comment (sudhakar-sah) [reply results/mcubench/benchmark_search_space_models.py]: Added command line arguments.

File: results/mcubench/benchmark_search_space_models.py (hunk 5 above)
Comment (sudhakar-sah) [reply results/mcubench/benchmark_search_space_models.py:160]: removed.

File: results/mcubench/benchmark_first_pareto_models.py (hunk 7 above)
Comment (sudhakar-sah) [reply results/mcubench/benchmark_first_pareto_models.py:161]: Fixed.

File: results/mcubench/one_for_all_mcubench_final.sh
Comment (sudhakar-sah) [reply results/mcubench/one_for_all_mcubench_final.sh:1]: replaced.

File: results/mcubench/benchmark_search_space_models.py (hunk 6 above)
Comment (sudhakar-sah) [reply results/mcubench/benchmark_search_space_models.py]: fixed.

File: results/mcubench/benchmark_search_space_models.py (hunk 4 above)
Comment (sudhakar-sah) [reply results/mcubench/benchmark_search_space_models.py:14]: removed.

**Review by sudhakar-sah — COMMENTED** [review]
Updated the code based on review changes.

**Pull request merged**